java/vector/src/main/java/org/apache/arrow/vector/BitVector.java

```diff
@@ -49,6 +50,34 @@ public BitVector(String name, BufferAllocator allocator) {
   }
 
   @Override
+  public void load(ArrowFieldNode fieldNode, ArrowBuf data) {
+    // When the vector is all nulls or all defined, the content of the buffer can be omitted
+    if (data.readableBytes() == 0 && fieldNode.getLength() != 0) {
+      data.release();
+      allocateNew(fieldNode.getLength());
+      int n = getSizeFromCount(fieldNode.getLength());
+      if (fieldNode.getNullCount() == 0) {
+        // all defined
+        // create an all 1s buffer
+        for (int i = 0; i < n; ++i) {
+          this.data.setByte(i, 0xFF);
```

Comment:
Will the superfluous 1's in the last byte (when the length is not a multiple of 8) cause any problems? e.g. in https://github.com/apache/arrow/pull/207 I noted that the test cases do not cover the case where the size is not a multiple of 8